runcoms/README.md

```diff
@@ -10,61 +10,199 @@ in */etc* by default.
+`#!/usr/bin/env zsh` or equivalent. 
+
+When launching a new terminal within an *X11* session, it is safe to 
+assume that you are *already logged in* so any terminal emulator will 
+launch an *interactive-non-login* shell. Which is, _per specification_ 
+the correct behavior, but not the behavior most users *expect*.
+
+
+There is a reason for this _specified_ behaviour:  `zlogin` files should 
+be used for *interaction oriented settings* such as custom completion, 
+keybindings, `tmux` session setup, and most `alias`es.
+
+The advantage of grouping all interaction oriented settings in 
+`.zlogin` is that non-interactive shells (executing scripts via 
+`zsh -c [script]`, the *shebang*, subshells or `make` invoked shells, 
+for example) will not be burdened by additional config settings making 
+them faster and more responsive, as _keybindings_ and _aliases_ are not 
+even loaded at startup, after all, an _executable non-interactive script_, 
+should never call on keybindings or aliases (assuming they were  properly
+written to be portable)
+
+
+Finally, it is a common source of frustration amongst graphical environment 
+users, to spend some time adding their customizations to `.zlogin` only 
+to see them ignored when launching a new `XTerm` or `tmux` pane. 
+
+This can be solved by calling (or aliasing) your terminal emulator with 
+the required option to invoke a *login* shell, for `XTerm`, for example 
+you would call `xterm -e zsh -ls` on tmux you may add one of the following 
+lines to your `.tmux.conf`:
+
+```config
+#tmux.conf:
+
+        set -g default-command 'exec /usr/bin/zsh -ls' #this will make zsh a login shell AND a session leader
+        set -g default-shell '/usr/bin/zsh -ls'
+```
+
+Here is an examplefor *i3wm* on ubuntu
+```bash
+
+#i3wm.conf:
+        ...
+        set $XTERM_CMD='xterm -e zsh -ls`
+        bind $mod+x $XTERM_CMD
+        ...
+```
+if you want to replace your non-login shell on your emulator with a brand-new 
+shiny *login* shell you can issue `exec zsh -ls` at your command prompt. 
+Doing this, however **will not write that terminal’s history to file** unless
+`INC_APPEND_HISTORY` was set when you strted the shell 
+
+`INC_APPEND_HISTORY` makes `zsh` append history entries as soon as the 
+command is finished. *zprezto* helpfuly sets `INC_APPEND_HISTORY`, 
+`INC_APPEND_HISTORY_TIME`, and `SHARE_HISTORY` which makes your history 
+available immediately from different terminals on command completion, 
+and records execution time on the history file. Pretty neat, huh?)
+
+You could tack-on all interactive customizations on `zshrc` but you will
+end up with bloated shells on non-interactive environments. Perhaps desktop 
+workstation users will not notice any difference, but in resource-constrained 
+environments it may be significant in terms of execution time and energy 
+consumption (v.gr when running a Raspberry PI on batteries or on a cell 
```

Comment:
`v.gr`?

Reply:
**v.gr** : shorthand of _verbi gratia_  / "[for example](https://en.wikipedia.org/wiki/List_of_Latin_phrases_(V)#verbi_gratia)" although literal latin translation would be "for the sake of a word"... she's **Q.E.D.**'s (Quod Erat Demostrandum) ugly cousin... :smiley_cat: